### DISCUSSION TITLE
Charon eats up to 5% CPU when idle

### DISCUSSION THREAD

**Discussion by bmansvk** [strongswan/strongswan#487] [Q&A]
I am running strongswan-5.9.1-1.el8.x86_64 from RHEL/EPEL. Charon process eats up to 5% CPU (measured using top). It is the same when strongswan is configured or when there is default configuration from RPM install (no connections configured). There is no connection, totally idle strongswan and it eats up to 5% CPU

It seems it is somehow interfering with my OS network configuration:
- when running in docker container with separate network there is not CPU usage
- when running in docker container with --net=host (host OS networking), there is up to 5% CPU usage

I tried to debug charon running from terminal with debug level incrased but found nothing interesting.

Do you have any idea what causes that CPU usage? How can I debug it?

Host OS networking has several bridges (podman/docker), wireguard tunnels, NIC use tagged VLANs.

**Comment by tobiasbrunner** [comment]
> I tried to debug charon running from terminal with debug level incrased but found nothing interesting.

How exactly did you configure [logging](https://wiki.strongswan.org/projects/strongswan/wiki/LoggerConfiguration)? The _knl_ subsystem would be a likely target, however, depending on the events generated by the kernel, there might not be any log messages.


> Host OS networking has several bridges (podman/docker), wireguard tunnels, NIC use tagged VLANs.

If there are a lot of changes (regarding routes, addresses or interfaces) this will produce a lot of events that are handled by the _kernel-netlink_ plugin (it keeps track of some of that and uses events to trigger roam events). You can see the kernel events with `ip monitor address link route rule` (listening to events for the last two can be disabled via `charon.process_route` and `charon.plugins.kernel-netlink.process_rules`, respectively). It's also possible to ignore certain interfaces (routing tables too in theory, but that filter is currently not applied when processing events, only when looking for source addresses).
